iceoryx_utils/include/iceoryx_utils/cxx/forward_list.hpp

```diff
@@ -0,0 +1,375 @@
+
+    // cpy c'tor by different CapacityOther-sized list, repeat for remaining copy/move ops
+    // template<uint64_t CapacityOther, typename=std::enable_if<CapacityOther <Capacity, void>::type>  // PRQA S 1051
+    // list(const list<T, CapacityOther> & rhs) noexcept;  // PRQA S 1051
+
+    /// @brief nested iterator class for list element operations including element access via dereferencing
+    ///         iterator may be assigned to different list (non-const pointer to m_List), different list iterators
+    ///         however are not compareable
+    class iterator
+    {
+      public:
+        // provide the following public types for a std::iterator_traits compatible iterator_category interface
+        using iterator_category = std::forward_iterator_tag;
+        using value_type = T;
+        using difference_type = void; // so far no differnce operations supported
+        using pointer = T*;
+        using reference = T&;
+        // end of iterator_traits interface
+
+        /// @brief prefix increment iterator, so it points to the next list element
+        ///         when trying to increment beyond the end of the list,
+        ///         iterator stays pointing at the end and a message is forwarded to the error_message
+        ///         handler / cerr stream
+        /// @return reference to this iterator object
+        iterator& operator++() noexcept;
+
+
+        /// @brief comparing list iterators on equality
+        ///         the referenced list position is compared, not the content of the list element (T-typed)
+        ///         there is no content for meta-elements like before_begin() and end()
+        ///         only iterators of the same parent list can be compared; in case of misuse, terminate() is invoked
+        /// @param[in] rhs_iter is the 2nd iterator to compare to
+        /// @return list position for two iterators is the same (true) or different (false)
+        bool operator==(const iterator rhs_iter) const noexcept;
+
+        /// @brief comparing list iterators on equality
+        ///         the referenced list position is compared, not the content of the list element (T-typed)
+        ///         there is no content for meta-elements like before_begin() and end()
+        ///         only iterators of the same parent list can be compared; in case of misuse, terminate() is invoked
+        /// @param[in] rhs_citer is the 2nd iterator to compare to
+        /// @return list position for two iterators is the same (true) or different (false)
+        bool operator==(const const_iterator rhs_citer) const noexcept;
+
+        /// @brief dereferencing element content via iterator-position element
+        /// @return reference to list element data
+        T& operator*() const noexcept;
+
+        /// @brief dereferencing element content via iterator-position element
+        /// @return pointer to list element data
+        T* operator->() const noexcept;
```

Comment:
This should not be a const method since you are able to change the underlying structure. Please change it into
` T* operator->() noexcept;`